sunxspex/emission.py

```diff
@@ -472,26 +476,50 @@ def integrate_part(*, model, photon_energies, maxfcn, rerr, eelow, eebrk, eehigh
     intsum = np.zeros_like(photon_energies, dtype=np.float64)
     ier = np.zeros_like(photon_energies)
 
+    if integrator is None:
+        integrator = gauss_legendre
+    elif not callable(integrator):
+        raise TypeError('integrator must ba a callable')
+
     # Copy indices over which to carry out the integration
     i = ll[:]
 
+    electron_dist = BrokenPowerLawElectronDistribution(p=p, q=q, eelow=eelow, eebrk=eebrk,
+                                                       eehigh=eehigh)
+
+    def model_func(x_log, *, model, electron_dist, photon_energy, p, q, eelow, eebrk, eehigh, z, efd):
+        mc2 = const.get_constant('mc2')
+        clight = const.get_constant('clight')
+
+        electron_energy = 10**x_log
+        brem_cross = bremsstrahlung_cross_section(electron_energy, photon_energy, z)
+        collision_loss = collisional_loss(electron_energy)
+        pc = np.sqrt(electron_energy * (electron_energy + 2.0 * mc2))
+
+        density = electron_dist.density(electron_energy)
+        if model == 'thick-target':
+            return (electron_energy * np.log(10) * density * brem_cross * pc
```

Comment:
Looks like this is the same as [line 408](https://github.com/sunpy/sunxspex/pull/64/files#diff-720f0f2daa949285d71390726972a7cb2130988f4faf4247c17d9cc76fba5052R408).  Perhaps it should be moved into it's own simple function, e.g. `_thick_target_func` so there's only one place it's defined.

Same for the thin target case.

Reply:
This can also be removed now